### DISCUSSION TITLE
Optional full text file search in Web UI

### DISCUSSION THREAD

**Discussion by EuroTrash2** [drakkan/sftpgo#1090] [Ideas]
This project is simply fantastic. It's the swiss army knife of self hosting files. Being filesystem based makes it so easy to backup and integrate with one's own scripts and tools.

I'd love to see a search bar in the web UI, to be able to perform a full text search. The implementation could use eg. Xapian or any other open source indexer.

One of the uses would be eg. quickly searching for documents on the server using one's mobile phone.

**Comment by drakkan** [comment]
Hi,

I'm not an expert of full text search libraries/engines. 
This could be implemented if the external full text engine provides an API to search in a particular path (the user's home directory) and return the full path to files that match,
SFTPGo supports several storage providers, I guess this will only work with local files with encryption disabled
